src/lang/KclSingleton.ts

```diff
@@ -293,6 +293,12 @@ export class KclManager {
       this._cancelTokens.delete(currentExecutionId)
       return
     }
+
+    // Exit sketch mode if the AST is empty
+    if (this._isAstEmpty(ast)) {
+      await this.disableSketchMode()
+    }
```

Comment:
This `executeAst` function is probably running when you first enter your first sketch, with an empty AST, and throwing a wrench in things from the jump in certain Playwright tests. I still agree with the approach, but does this check maybe need to live in a different place?

Reply:
Thanks for the tip, I'll take a look at this.